kpi/deployment_backends/mock_backend.py

```diff
@@ -201,6 +205,53 @@ def duplicate_submission(
         settings.MONGO_DB.instances.insert_one(duplicated_submission)
         return duplicated_submission
 
+    def get_attachment_content(self, user, submission_uuid, response_xpath):
+
+        try:
+            submission_xml = self.get_submissions(
+                user, format_type=SUBMISSION_FORMAT_TYPE_XML, query={
+                    '_uuid': submission_uuid
+                }
+            )
+            first_submission = submission_xml[0]
+        except StopIteration:
+            raise Http404
+
+        submission_tree = ET.ElementTree(
+            ET.fromstring(first_submission)
+        )
+        response_element = submission_tree.find(response_xpath)
+        try:
+            response_filename = response_element.text
+        except AttributeError:
+            raise InvalidXPathException
+
+        try:
+            submission_json = next(
+                iter(
+                    self.get_submissions(
+                        user, format_type=SUBMISSION_FORMAT_TYPE_JSON, query={
+                            '_uuid': submission_uuid
+                        }
+                    )
+                )
+            )
+        except StopIteration:
+            raise Exception('No matching submission')
```

Comment:
Please apply the same changes that you did for the XML version (i.e: No need to use `next(iter())` on a list)